### ISSUE TITLE
FreeBayes is currently not supported

### ISSUE THREAD

**Issue by FriederikeHanssen** [qbic-pipelines/vcftomaf#2]
### Description of feature

I noticed a couple of things when testing with tumor-only samples:

1) The used `--vcf-tumor-id` is not correctly resolved:

Sample sheet has: `QFEND268A3,QFEND268A3,` as `sample,tumor_id` . For some reason this is what is in the `.command.sh`:
```
    --tumor-id QFEND286A1   --vcf-tumor-id  QFEND286A1  \
    --input-vcf QFEND268A3.vcf \
    --output-maf QFEND268A3.maf
 ```
but the actual `--vcf-tumor-id  QFEND286A1` would be `NGS02QFEND268A3_NGS02QFEND268A3` , so a) it is not correctly resolved from the given information and b) somehow the last digit is changed. On further inspection it has turned out that in general the ids are completey swapped occasionally. I guess some channel magic doesn't work somewhere, i.e. here:
```
    --tumor-id QFEND282A3   --vcf-tumor-id  QFEND282A3  \
    --input-vcf QFEND270AE.vcf \
    --output-maf QFEND270AE.maf
```

2) Freebayes vcfs did not work for me, but resulted in: `Command error:
  [E::vcf_parse_format] Invalid character '.' in 'GQ' FORMAT field at chr1:10146
  Error: VCF parse error
` 
3) There is an unused `bcftools stats` module

**Comment by FriederikeHanssen** [comment]
1. and 3. have been addressed by PR #3

**Comment by famosab** [comment]
2. Is probably related to the way the 'GQ' FORMAT field is declared (see https://github.com/samtools/bcftools/issues/420). Maybe this can be solved with `bcftools reheader`

**Comment by famosab** [comment]
The error probably does not occur anymore since newer freebayes VCFs declare the GQ FORMAT field like
```
##FORMAT=<ID=GQ,Number=1,Type=Float,Description="Genotype Quality, the Phred-scaled marginal (or unconditional) probability of the called genotype">
```
which marks '.' as valid character.

**Comment by famosab** [comment]
However @LeonHornich observed that running the pipeline with samples coming from freebayes leads to empty MAF files.

**Comment by LeonHornich** [comment]
> However @LeonHornich observed that running the pipeline with samples coming from freebayes leads to empty MAF files.

Just for clarification, the .maf files are not empty. The version and the maf headers are still being printed. The files then look like this:

```
#version 2.4
Hugo_Symbol	Entrez_Gene_Id	Center	NCBI_Build	Chromosome	Start_Position	End_Position	Strand	...

```

**Comment by LeonHornich** [comment]
I checked my pipeline runs again. Looks like my vcf files from `bcftools` already don't have variant entries. In a similar pattern to the maf files, the `bcftools` produced files include the info rows and vcf headers but no entries. So we might be looking at a issue around `bcftools` which is being called prior to `vcf2maf`.

**Comment by famosab** [comment]
Seems to be the same case for the test data. I guess there is something off with the filtering or the channels. Investigating now.

**Comment by famosab** [comment]
We do filter all variants by `PASS` with `bcftools view`. The test data do not have a `PASS` field (looking for VCFs with those fields now). And @LeonHornich seems that your VCFs also do not have `PASS` fields.
Removing the `PASS` filtering leads to filled MAFs for the testdata.

**Comment by famosab** [comment]
From what I have tested with @LeonHornich  files coming from Freebayes it seems that the original error which was described by @FriederikeHanssen is no longer there. Closing this issue for now as optional filtering by PASS was implemented in #18. Feel free to re-open if the bug comes back :)
